snapshots/devmapper/pool_device.go

```diff
@@ -479,6 +481,41 @@ func (p *PoolDevice) GetUsage(deviceName string) (int64, error) {
 	return count * dmsetup.SectorSize, nil
 }
 
+func (p *PoolDevice) IsOutOfSpace(deviceName string) (bool, error) {
+	var (
+		transactionId uint64
+		dataUsed      uint64
+		dataTotal     uint64
+		metadataUsed  uint64
+		metadataTotal uint64
+		lowWaterMark  uint64
+	)
+
+	status, err := dmsetup.Status(deviceName)
+	if err != nil {
+		return false, errors.Wrapf(err, "can't get status for device %q", deviceName)
+	}
+
+	if len(status.Params) == 0 {
+		return false, errors.Errorf("failed to get the number of used blocks, unexpected output from dmsetup status")
+	}
+
+	fmt.Sscanf(strings.Join(status.Params, " "), "%d %d/%d %d/%d", &transactionId, &metadataUsed, &metadataTotal, &dataUsed, &dataTotal)
+
+	table, err := dmsetup.Table(deviceName)
+	if err != nil {
+		return false, errors.Wrapf(err, "can't get table for device %q", deviceName)
+	}
+
+	lowWaterMarkFromTable := strings.Split(table, " ")[6]
+	fmt.Sscanf(lowWaterMarkFromTable, "%d", &lowWaterMark)
+
+	if dataTotal-dataUsed < lowWaterMark {
+		return true, err
+	}
+	return false, err
```

Comment:
I think `return true, nil` / `return false, nil` is cleaner.

Reply:
Thanks. Done.